src/js/newTab.js

```diff
@@ -1,24 +1,41 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
-/*globals gGrid, gPage, gStrings, gUpdater, gUserDatabase, gPinnedLinks, gBlockedLinks, async*/
+/*globals Request, CacheTasks, gGrid, gPage, gUpdater, gUserDatabase,
+  gPinnedLinks, gBlockedLinks, async*/
 
 "use strict";
 
 (function(exports) {
   const gNewTab = {
-    _prefsObjectStoreKeys: {"pinnedLinks": [], "blockedLinks": []},
 
     listeners: {},
 
+    _l10nStrings: new Map(),
+
     init() {
-      // Add a listener for messages sent from the browser.
-      // The listener calls our associated callback functions.
-      window.addEventListener("message", message => {
-        for (let callback of this.listeners[message.data.name]) {
-          callback(message.data.data);
+      return async.task(function*() {
+        // Add a listener for messages sent from the browser.
+        // The listener calls our associated callback functions.
+        window.addEventListener("message", message => {
+          for (let callback of this.listeners[message.data.name]) {
+            callback(message.data.data);
+          }
+        });
+        let json = {};
+        try {
+          let request = new Request("./locale/strings.json");
```

Comment:
Over here, we assume that the locale directory will be local to newTab.js, correct?
We'll need to grab the locale from the URL in another issue.